core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -71,7 +71,7 @@ class BrokerServer(
   val initialOfflineDirs: Seq[String],
 ) extends KafkaBroker {
   val threadNamePrefix = sharedServer.threadNamePrefix
-  val config = sharedServer.config
+  val config = new KafkaConfig(sharedServer.sharedServerConfig.props, false, None)
```

Comment:
Could we add a method on SharedServer like `createKafkaConfig` rather than expose `sharedServerConfig` to BrokerServer/ControllerServer? I worry someone down the line might read from the shared config rather than the local one thats getting dynamic updates.

Reply:
Good idea